src/App.tsx

```diff
@@ -15,36 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetic = 1,
+  length = 2,
+}
+
+export const sortGoods = (goods, params) => {
+  const { sortBy, isReversed } = params;
+  const sortedGoods = [...goods];
+
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.alphabetic:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          throw new Error('There is no such type of sort');
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(null);
+  const [isReversed, setReversed] = useState(false);
```

Comment:
The `useState` hook for `isReversed` should have an initial type specified. Consider using `useState<boolean>(false)` to ensure type safety.

Reply:
> The `useState` hook for `isReversed` should have an initial type specified. Consider using `useState<boolean>(false)` to ensure type safety.

In this case it's ok, typescript understands that it's boolean